keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/TimeWindowConstraintEvaluator.kt

```diff
@@ -0,0 +1,203 @@
+
+      if (hoursMatch && daysMatch) {
+        return true
+      }
+    }
+
+    return false
+  }
+
+  private fun parseHours(hourConfig: String?, deliveryConfig: DeliveryConfig, targetEnvironment: String): Set<Int> {
+    if (hourConfig == null) {
+      return emptySet()
+    }
+
+    val hours = mutableSetOf<Int>()
+    val trimmed = hourConfig.replace(whiteSpace, "")
+    val elements = trimmed.split(",")
+
+    elements.forEach {
+      when {
+        it.isInt() -> hours.add(it.toInt())
+        it.isIntRange() -> hours.addAll(it.hourRange())
+        else -> throw InvalidConstraintException(
+          "Invalid time-window constraint ($it) on deliveryConfig: ${deliveryConfig.name}, " +
+            "application: ${deliveryConfig.application}, environment: $targetEnvironment")
+      }
+    }
+
+    return hours
+  }
+
+  private fun parseDays(dayConfig: String?, deliveryConfig: DeliveryConfig, targetEnvironment: String): Set<String> {
+    if (dayConfig == null) {
+      return emptySet()
+    }
+
+    val days = mutableSetOf<String>()
+    val trimmed = dayConfig.replace(whiteSpace, "")
+      .toLowerCase()
+    val elements = trimmed.split(",")
+
+    elements.forEach {
+      when {
+        it.isDay() -> days.add(it)
+        it.isDayRange() -> days.addAll(it.dayRange())
+        else -> throw InvalidConstraintException(
+          "Invalid time-window constraint ($it) on deliveryConfig: ${deliveryConfig.name}, " +
+            "application: ${deliveryConfig.application}, environment: $targetEnvironment")
+      }
+    }
+
+    return days
+  }
+
+  private fun String.isInt(): Boolean = this.matches(intOnly)
+
+  private fun String.isIntRange(): Boolean = this.matches(intRange)
+
+  private fun String.hourRange(): Set<Int> {
+    val hours = this.split("-")
+    return IntRange(hours[0].toInt(), hours[1].toInt())
+      .toSet()
+  }
+
+  private fun String.isDay(): Boolean = daysOfWeek.contains(this)
+
+  private fun String.isDayRange(): Boolean {
+    val days = this.split("-")
+    if (days.size != 2) {
+      return false
+    }
+
+    return daysOfWeek.contains(days[0]) && daysOfWeek.contains(days[1])
+  }
+
+  private fun String.dayRange(): Set<String> {
+    /**
+     * Convert Mon-Fri or Monday-Friday to [DayOfWeek] integers to compute
+     * a range, then back to a set of individual days that today can be
+     * matched against.
+     */
+    val days = this.split("-").map { it.capitalize() }
+    val day1 = fullDayFormatter.parseUnresolved(days[0], ParsePosition(0))
+      ?: shortDayFormatter.parseUnresolved(days[0], ParsePosition(0))
+      ?: throw InvalidConstraintException("Failed parsing day range $this")
+    val day2 = fullDayFormatter.parseUnresolved(days[1], ParsePosition(0))
+      ?: shortDayFormatter.parseUnresolved(days[1], ParsePosition(0))
+      ?: throw InvalidConstraintException("Failed parsing day range $this")
+
+    val intRange = IntRange(DayOfWeek.from(day1).value, DayOfWeek.from(day2).value)
```

Comment:
What would happen here if you specify something backwards like `thu-mon`?

Reply:
Probably not what the user would want.

Reply:
Should we throw an `InvalidConstraintException` in that case, then?

Reply:
Should we throw an `InvalidConstraintException` in that case, then?

Reply:
Updated the behavior so wrap around hours and days work as one would reasonably expect.